### PR TITLE
Manually sync AK/3.4

### PR DESCRIPTION
**Pull request by jolshan**
Jenkins job had conflict, so manual merge was done here.

	both modified:   core/src/main/scala/kafka/server/SharedServer.scala —> trivial conflict
	both modified:   tests/kafkatest/tests/streams/streams_upgrade_test.py —> take ak change, will run system test

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/server/SharedServer.scala [hunk 1]
```diff
@@ -97,6 +101,10 @@ class SharedServer(
   @volatile var raftManager: KafkaRaftManager[ApiMessageAndVersion] = _
   @volatile var brokerMetrics: BrokerServerMetrics = _
   @volatile var controllerMetrics: QuorumControllerMetrics = _
+  @volatile var loader: MetadataLoader = _
+  val snapshotsDiabledReason = new AtomicReference[String](null)
```
Comment (ijuma) [inline core/src/main/scala/kafka/server/SharedServer.scala:105]: Is this typo upstream too?

**Review by ijuma — APPROVED** [review]
LGTM, thanks.

**Comment by jolshan** [comment]
Confirmed the streams test passes with the new change. 👍

File: core/src/main/scala/kafka/server/SharedServer.scala (hunk 1 above)
Comment (jolshan) [reply core/src/main/scala/kafka/server/SharedServer.scala:105]: Yup -- https://github.com/apache/kafka/blob/9089a4ddb9cbde195862514f027ced594883b2d9/core/src/main/scala/kafka/server/SharedServer.scala#L105

I'll leave it for now and we can fix upstream (its also in trunk)

**Comment by jolshan** [comment]
Test failures look flaky. Will continue with merge.

**Pull request merged**